### ISSUE TITLE
add station metadata JSON microspecification

### ISSUE THREAD

**Issue by tomkralidis** [wmo-im/wis2box#54]
The below represents the lifecycle of WIGOS station metadata in wis2node:

### Caching (current)
- wis2node caches OSCAR/Surface station metadata from an operator generated CSV of WSIs in order to process data accordingly
  - `wis2node metadata station cache /path/to/station_list.csv`
### Publishing (future/planned)
- wis2node publishes updated observation metadata to OSCAR/Surface M2M API
  - `wis2node metadata station publish /path/to/station_list.csv`

wis2node uses [pyoscar](https://github.com/wmo-cop/pyoscar) as the Python package to interact with OSCAR/Surface.

pyoscar allows for obtaining station information in either WMDR or JSON.  wis2node uses the latter, which is an "internal"-ish representation which wis2node should not depend on for the long term.  Using the pyoscar CLI as an example:

```bash
# get a single station by WIGOS identifier in bespoke JSON format by default
pyoscar station --identifier 0-20000-0-71151

# get a single station by WIGOS identifier in WIGOS XML format
pyoscar station --identifier 0-20000-0-71151 --format=XML
```

In summary, wis2node is caching OSCAR/Surface metadata using the bespoke JSON format.  We should access WMDR and then translate to an internal JSON representation for wis2node/csv2bufr processing.

cc @efucile

**Comment by david-i-berry** [comment]
At the moment something very simple would work for the csv2bufr component, to get started I'd suggest something like:

```
{
  "$id": "wmdr.csv2bufr.wis2.0.node.wis",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
    "type": "object",
    "properties": {
      "id": {
        "type": "string",
        "description": "Unique id for record"
      },
      "type":{
        "type": "string",
        "description": "geojson feature type, only Features supported",
        "enum": ["Feature"]
      },
      "geometry": {
        "type": "object",
        "properties": {
          "type": {
            "type": "string",
            "description": "geometry type, only 3d point supported. Future versions will need to support additional types",
            "enum": ["Point"]
          },
          "coordinates": {
            "type": "array",
            "items": {
              "type": "number"
            },
            "minItems": 3,
            "maxItems": 3,
            "description": "[longitude, latitude, elevation] of station (CRS: WGS84)"
          }
        }
      },
      "properties": {
        "type": "object",
        "properties": {
          "wigos_id_series": {
            "type": "integer",
            "description": "WIGOS identifer - series number"
          },
          "wigos_id_issuer": {
            "type": "integer",
            "description": "WIGOS identifer - issuer"
          },
          "wigos_id_issue_number": {
            "type": "integer",
            "description": "WIGOS identifer - issue number"
          },
          "wigos_id_local_identifier": {
            "type": "string",
            "description": "WIGOS identifer - local identifier"},
          "station_name": {
            "type": "string",
            "description": "Name of station, platform or facility"
          }
        }
      }
    },
  "valid_from": {
    "type": "string",
    "description": "date record is valid from"
  },
  "valid_to": {
    "type": "string",
    "description": "date record is valid from"
  },
  "_meta": {
    "type": "object",
    "properties": {
      "cache_date": {
        "type": "string",
        "description": "date record was last downloaded"
      },
      "download_url": {
        "type": "string",
        "description": "URL where data was downloaded from"
      }
    }
  }
}
```

We have all of the above elements in the CSV data file as well but at lower numerical precision. We could request that these are changed and drop dependence on the data from OSCAR.
